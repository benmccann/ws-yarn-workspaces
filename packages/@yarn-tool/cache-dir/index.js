"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getYarnCacheDir = void 0;
const upath2_1 = require("upath2");
const fs_extra_1 = require("fs-extra");
const cross_spawn_extra_1 = require("cross-spawn-extra");
function getYarnCacheDir() {
    try {
        let cp = cross_spawn_extra_1.sync('yarn', [
            'config',
            'current',
            '--json',
        ], {
            stripAnsi: true,
            env: process.env,
        });
        let data = JSON.parse(JSON.parse(cp.stdout.toString()).data);
        if (data.tempFolder) {
            return upath2_1.normalize(data.tempFolder);
        }
    }
    catch (e) {
    }
    if (process.env.YARN_CACHE_FOLDER && fs_extra_1.pathExistsSync(process.env.YARN_CACHE_FOLDER)) {
        return upath2_1.normalize(process.env.YARN_CACHE_FOLDER);
    }
}
exports.getYarnCacheDir = getYarnCacheDir;
exports.default = getYarnCacheDir;
//# sourceMappingURL=index.js.map