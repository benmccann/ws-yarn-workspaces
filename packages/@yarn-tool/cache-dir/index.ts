import { join, normalize } from "upath2";
import { pathExistsSync } from 'fs-extra';
import { sync as crossSpawnExtra } from 'cross-spawn-extra';

export function getYarnCacheDir(): string
{
	try
	{
		let cp = crossSpawnExtra('yarn', [
			'config',
			'current',
			'--json',
		], {
			stripAnsi: true,
			env: process.env,
		});

		let data = JSON.parse(JSON.parse(cp.stdout.toString()).data);

		if (data.tempFolder)
		{
			return normalize(data.tempFolder)
		}
	}
	catch (e)
	{

	}

	if (process.env.YARN_CACHE_FOLDER && pathExistsSync(process.env.YARN_CACHE_FOLDER))
	{
		return normalize(process.env.YARN_CACHE_FOLDER)
	}
}

export default getYarnCacheDir
