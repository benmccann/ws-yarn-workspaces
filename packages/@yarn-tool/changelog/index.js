"use strict";
/**
 * Created by user on 2020/6/15.
 */
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
tslib_1.__exportStar(require("./lib/types"), exports);
tslib_1.__exportStar(require("./lib/lerna"), exports);
tslib_1.__exportStar(require("./lib/yargs-setting"), exports);
const lerna_1 = require("./lib/lerna");
exports.default = lerna_1.updateChangelogByCwd;
//# sourceMappingURL=index.js.map