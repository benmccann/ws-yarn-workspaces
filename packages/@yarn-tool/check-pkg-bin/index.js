"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const pkg_1 = require("./lib/pkg");
(0, tslib_1.__exportStar)(require("./lib/core"), exports);
(0, tslib_1.__exportStar)(require("./lib/ws"), exports);
(0, tslib_1.__exportStar)(require("./lib/pkg"), exports);
exports.default = pkg_1.checkPkgDir;
//# sourceMappingURL=index.js.map