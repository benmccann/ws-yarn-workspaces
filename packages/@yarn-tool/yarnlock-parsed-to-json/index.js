"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.yarnLockParsedToRawJSON = exports.yarnLockParsedV2ToRawJSON = exports.yarnLockParsedV1ToRawJSON = void 0;
const yarnlock_error_1 = require("@yarn-tool/yarnlock-error");
const yarnlock_parse_assert_1 = require("@yarn-tool/yarnlock-parse-assert");
function yarnLockParsedV1ToRawJSON(parsedObject) {
    return {
        ...parsedObject.meta,
        object: parsedObject.data,
    };
}
exports.yarnLockParsedV1ToRawJSON = yarnLockParsedV1ToRawJSON;
function yarnLockParsedV2ToRawJSON(parsedObject) {
    return {
        __metadata: parsedObject.meta,
        ...parsedObject.data,
    };
}
exports.yarnLockParsedV2ToRawJSON = yarnLockParsedV2ToRawJSON;
function yarnLockParsedToRawJSON(parsedObject, options) {
    (0, yarnlock_parse_assert_1.assertYarnLockParsed)(parsedObject);
    if ((0, yarnlock_parse_assert_1.isYarnLockParsedV2)(parsedObject)) {
        return yarnLockParsedV2ToRawJSON(parsedObject);
    }
    else if ((0, yarnlock_parse_assert_1.isYarnLockParsedV1)(parsedObject)) {
        return yarnLockParsedV1ToRawJSON(parsedObject);
    }
    if (options === null || options === void 0 ? void 0 : options.throwError) {
        throw (0, yarnlock_error_1.newYarnLockParsedVersionError)();
    }
}
exports.yarnLockParsedToRawJSON = yarnLockParsedToRawJSON;
exports.default = yarnLockParsedToRawJSON;
//# sourceMappingURL=index.js.map