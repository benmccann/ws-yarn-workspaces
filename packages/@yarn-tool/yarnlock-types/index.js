"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumLanguageName = exports.EnumLinkType = exports.EnumYarnLockSourceV1Type = exports.EnumDetectYarnLockInputType = exports.EnumDetectYarnLock = void 0;
var EnumDetectYarnLock;
(function (EnumDetectYarnLock) {
    EnumDetectYarnLock[EnumDetectYarnLock["v1"] = 1] = "v1";
    EnumDetectYarnLock[EnumDetectYarnLock["v2"] = 2] = "v2";
    /**
     * @deprecated do not use this if u want check is version is berry
     * @type {EnumDetectYarnLock.berry}
     */
    EnumDetectYarnLock[EnumDetectYarnLock["berry"] = 2] = "berry";
    EnumDetectYarnLock[EnumDetectYarnLock["v3"] = 3] = "v3";
    /**
     * @deprecated do not use this if u want check is version is berry
     * @alias {EnumDetectYarnLock.v3}
     */
    EnumDetectYarnLock[EnumDetectYarnLock["canary"] = 3] = "canary";
    EnumDetectYarnLock[EnumDetectYarnLock["unknown"] = 0] = "unknown";
})(EnumDetectYarnLock || (exports.EnumDetectYarnLock = EnumDetectYarnLock = {}));
var EnumDetectYarnLockInputType;
(function (EnumDetectYarnLockInputType) {
    EnumDetectYarnLockInputType[EnumDetectYarnLockInputType["unknown"] = 0] = "unknown";
    EnumDetectYarnLockInputType[EnumDetectYarnLockInputType["simple"] = 1] = "simple";
    EnumDetectYarnLockInputType[EnumDetectYarnLockInputType["parse_raw"] = 2] = "parse_raw";
})(EnumDetectYarnLockInputType || (exports.EnumDetectYarnLockInputType = EnumDetectYarnLockInputType = {}));
var EnumYarnLockSourceV1Type;
(function (EnumYarnLockSourceV1Type) {
    EnumYarnLockSourceV1Type["success"] = "success";
    EnumYarnLockSourceV1Type["merge"] = "merge";
    EnumYarnLockSourceV1Type["conflict"] = "conflict";
})(EnumYarnLockSourceV1Type || (exports.EnumYarnLockSourceV1Type = EnumYarnLockSourceV1Type = {}));
var EnumLinkType;
(function (EnumLinkType) {
    EnumLinkType["hard"] = "hard";
    EnumLinkType["soft"] = "soft";
})(EnumLinkType || (exports.EnumLinkType = EnumLinkType = {}));
var EnumLanguageName;
(function (EnumLanguageName) {
    EnumLanguageName["node"] = "node";
    EnumLanguageName["unknown"] = "unknown";
})(EnumLanguageName || (exports.EnumLanguageName = EnumLanguageName = {}));
//# sourceMappingURL=index.js.map