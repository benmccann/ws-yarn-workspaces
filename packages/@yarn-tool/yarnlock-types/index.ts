import { IDependency } from '@ts-type/package-dts/lib/package-json/types';
import { ITSTypeAndStringLiteral } from 'ts-type/lib/helper/string';

export type IYarnLockParsed = IYarnLockParsedV1 | IYarnLockParsedV2;

/**
 * use for avoid typescript show error
 */
export interface IYarnLockParsedLoose<D extends IYarnLockDataRowBase = IYarnLockDataRowBase, K extends string = string>
{
	verType: EnumDetectYarnLock,
	meta?: Record<string, any>,
	data: IYarnLockDataRecord<D, K>,
}

export interface IYarnLockDataRowBase
{
	version: string
	/**
	 * 依賴列表
	 */
	dependencies?: IDependency,
}

export interface IYarnLockDataRowV1 extends IYarnLockDataRowBase
{
	/**
	 * 安裝來源網址
	 */
	resolved?: string,
	/**
	 * hash key
	 */
	integrity?: string,
}

export interface IYarnLockDataRowV2 extends IYarnLockDataRowBase
{
	resolution: string,
	checksum?: string,
	languageName?: string | ITSTypeAndStringLiteral<EnumLanguageName>,
	linkType?: string | ITSTypeAndStringLiteral<EnumLinkType>,
}

export type IYarnLockDataRecord<D extends IYarnLockDataRowBase = IYarnLockDataRowBase, K extends string = string> = Record<K, D>

export enum EnumDetectYarnLock
{
	v1 = 1,
	v2 = 2,
	/**
	 * @deprecated do not use this if u want check is version is berry
	 * @type {EnumDetectYarnLock.berry}
	 */
	berry = 2,
	v3 = 3,
	/**
	 * @deprecated do not use this if u want check is version is berry
	 * @alias {EnumDetectYarnLock.v3}
	 */
	canary = 3,
	unknown = 0,
}

export const enum EnumDetectYarnLockInputType
{
	unknown,
	simple,
	parse_raw,
}

export type IEnumDetectYarnLockValues = typeof EnumDetectYarnLock[keyof typeof EnumDetectYarnLock];
export type IValidYarnLockVersion = Exclude<IEnumDetectYarnLockValues, EnumDetectYarnLock.unknown>;
export type IYarnLockVersionSyml = Exclude<IValidYarnLockVersion, EnumDetectYarnLock.v1>;
export type IYarnLockMetaVersion = `${number}`

export interface IYarnLockParsedV1<D extends IYarnLockDataRecord<IYarnLockDataRowBase> = IYarnLockDataRecord<IYarnLockDataRowV1>>
{
	verType: EnumDetectYarnLock.v1,
	meta?: {
		type?: string | 'success' | 'merge' | 'conflict'
		version?: never,
	},
	data: D,
}

export interface IYarnLockParsedV2<D extends IYarnLockDataRecord<IYarnLockDataRowBase> = IYarnLockDataRecord<IYarnLockDataRowV2>>
{
	verType: IYarnLockVersionSyml,
	meta?: {
		type?: never
		version?: string,
		cacheKey?: string,
	},
	data: D,
}

export type IUnpackYarnLockDataRow<T extends IYarnLockParsedV1 | IYarnLockParsedV2> = T extends {
		data: IYarnLockDataRecord<infer D>,
	}
	? D extends IYarnLockDataRowV2
		? D
		: D extends IYarnLockDataRowV1
			? D
			: D extends IYarnLockDataRowBase
				? D
				: never
	: never
	;

export const enum EnumYarnLockSourceV1Type
{
	'success' = 'success',
	'merge' = 'merge',
	'conflict' = 'conflict',
}

/**
 * yarn.lock v1
 * @see @yarnpkg/lockfile
 */
export interface IYarnLockRawSourceV1
{
	type?: ITSTypeAndStringLiteral<EnumYarnLockSourceV1Type>;
	object: IYarnLockDataRecord<IYarnLockDataRowV1>;
}

export type IYarnLockRawSourceV1Full = Required<IYarnLockRawSourceV1>

/**
 * yarn.lock v2
 * @see @yarnpkg/parsers
 */
export type IYarnLockRawSourceV2 = {
	__metadata?: {
		version?: string,
		cacheKey?: string,
	}
} & IYarnLockDataRecord<IYarnLockDataRowV2>;

export type IYarnLockSource = IYarnLockRawSourceV1 | IYarnLockRawSourceV2;

export type IYarnLockRawSourceV2Full = Required<IYarnLockRawSourceV2>

export enum EnumLinkType
{
	'hard' = 'hard',
	'soft' = 'soft',
}

export enum EnumLanguageName
{
	'node' = 'node',
	'unknown' = 'unknown',
}
