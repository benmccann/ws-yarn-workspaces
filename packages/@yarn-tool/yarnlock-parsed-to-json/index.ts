import { newYarnLockParsedVersionError } from '@yarn-tool/yarnlock-error';
import { assertYarnLockParsed, isYarnLockParsedV1, isYarnLockParsedV2 } from '@yarn-tool/yarnlock-parse-assert';
import {
	IYarnLockParsed,
	IYarnLockParsedV1,
	IYarnLockParsedV2,
	IYarnLockSource,
	IYarnLockRawSourceV1,
	IYarnLockRawSourceV2,
} from '@yarn-tool/yarnlock-types';

export function yarnLockParsedV1ToRawJSON(parsedObject: IYarnLockParsedV1): IYarnLockRawSourceV1
{
	return {
		...parsedObject.meta,
		object: parsedObject.data,
	} as any
}

export function yarnLockParsedV2ToRawJSON(parsedObject: IYarnLockParsedV2): IYarnLockRawSourceV2
{
	return {
		__metadata: parsedObject.meta,
		...parsedObject.data,
	} as any
}

export interface IOptionsYarnLockParsedToRawJSON
{
	throwError?: boolean,
}

export function yarnLockParsedToRawJSON(parsedObject: IYarnLockParsedV1,
	options?: IOptionsYarnLockParsedToRawJSON,
): IYarnLockRawSourceV1
export function yarnLockParsedToRawJSON(parsedObject: IYarnLockParsedV2,
	options?: IOptionsYarnLockParsedToRawJSON,
): IYarnLockRawSourceV2
export function yarnLockParsedToRawJSON(parsedObject: IYarnLockParsed,
	options?: IOptionsYarnLockParsedToRawJSON,
): IYarnLockSource
export function yarnLockParsedToRawJSON(parsedObject: IYarnLockParsed, options?: IOptionsYarnLockParsedToRawJSON)
{
	assertYarnLockParsed(parsedObject);

	if (isYarnLockParsedV2(parsedObject))
	{
		return yarnLockParsedV2ToRawJSON(parsedObject)
	}
	else if (isYarnLockParsedV1(parsedObject))
	{
		return yarnLockParsedV1ToRawJSON(parsedObject)
	}

	if (options?.throwError)
	{
		throw newYarnLockParsedVersionError();
	}
}

export default yarnLockParsedToRawJSON
