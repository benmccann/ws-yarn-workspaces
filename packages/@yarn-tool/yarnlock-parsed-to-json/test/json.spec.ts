//@noUnusedParameters:false

import { basename, extname } from 'path';
import { EnumDetectYarnLock, IYarnLockParsed, IYarnLockParsedV1, IYarnLockParsedV2 } from '@yarn-tool/yarnlock-types';
import { yarnLockParsedToRawJSON, yarnLockParsedV1ToRawJSON, yarnLockParsedV2ToRawJSON } from '../index';

beforeAll(async () =>
{

});

describe(basename(__filename, extname(__filename)), () =>
{
	const parsedObject: IYarnLockParsed = {
		verType: void 0,
		meta: {
			version: '5',
		},
		data: {

		}
	}

	test(`yarnLockParsedV1ToRawJSON`, () =>
	{
		const json: IYarnLockParsedV1 = {
			...(parsedObject as any as IYarnLockParsedV1),
			verType: EnumDetectYarnLock.v1,
		};

		let actual = yarnLockParsedV1ToRawJSON(json);

		expect(actual).toMatchSnapshot();

	});

	test(`yarnLockParsedV2ToRawJSON`, () =>
	{
		const json: IYarnLockParsedV2 = {
			...(parsedObject as any as IYarnLockParsedV2),
			verType: EnumDetectYarnLock.v2,
		};

		let actual = yarnLockParsedV2ToRawJSON(json);

		expect(actual).toMatchSnapshot();

	});

	test(`throw`, () => {

		expect(() => yarnLockParsedToRawJSON(parsedObject)).toThrowError();

		expect(() => yarnLockParsedToRawJSON({
			...(parsedObject as any as IYarnLockParsedV1),
			verType: EnumDetectYarnLock.v1,
		})).not.toThrowError();

		expect(() => yarnLockParsedToRawJSON({
			...(parsedObject as any as IYarnLockParsedV2),
			verType: EnumDetectYarnLock.v2,
		})).not.toThrowError();

	});

})
