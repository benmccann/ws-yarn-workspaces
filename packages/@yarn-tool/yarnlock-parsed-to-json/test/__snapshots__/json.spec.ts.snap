// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json.spec yarnLockParsedV1ToRawJSON 1`] = `
Object {
  "object": Object {},
  "version": "5",
}
`;

exports[`json.spec yarnLockParsedV2ToRawJSON 1`] = `
Object {
  "__metadata": Object {
    "version": "5",
  },
}
`;
